### PR TITLE
[KEYCLOAK-5958] - An obsolete Identity model is used in code snippets

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

File: authorization_services/topics/policy-js-policy.adoc [hunk 1]
```diff
@@ -42,28 +42,41 @@ if (contextAttributes.containsValue('kc.client.network.ip_address', '127.0.0.1')
 }
 ```
 
-You can also use role-based access control (RBAC):
+You can also use Role-Based Access Control (RBAC) in your policies. It is possible to check for
+any *realm* role granted to an user:
```
Comment (matthewhelmke) [inline authorization_services/topics/policy-js-policy.adoc]: You can also use Role-Based Access Control (RBAC) in your policies. Use this to determine which, if any, *realm* role has been granted to an user:

File: authorization_services/topics/policy-js-policy.adoc [hunk 2]
```diff
@@ -42,28 +42,41 @@ if (contextAttributes.containsValue('kc.client.network.ip_address', '127.0.0.1')
...
 
-if (identity.hasRole('keycloak_user')) {
+if (identity.hasRealmRole('keycloak_user')) {
     $evaluation.grant();
 }
 ```
 
-Or a combination of several access control mechanisms:
+Or even any *client* role granted to an user:
+
```
Comment (matthewhelmke) [inline authorization_services/topics/policy-js-policy.adoc]: Use this to determine which, if any, *client* role has been granted to an user:

File: authorization_services/topics/policy-js-policy.adoc [hunk 3]
```diff
@@ -42,28 +42,41 @@ if (contextAttributes.containsValue('kc.client.network.ip_address', '127.0.0.1')
...
 var email = attributes.getValue('email').asString(0);
 
-if (identity.hasRole('admin') || email.endsWith('@keycloak.org')) {
+if (identity.hasRealmRole('admin') || email.endsWith('@keycloak.org')) {
     $evaluation.grant();
 }
 ```
-
-When writing your own rules, keep in mind that the *$evaluation* object is an object implementing *org.keycloak.authorization.policy.evaluation.Evaluation*. For more information about what you can access from this interface, see the <<_policy_evaluation_api, Evaluation API>>.
+[NOTE]
```
Comment (matthewhelmke) [inline authorization_services/topics/policy-js-policy.adoc]: NOTE: When writing your own rules, ...

**Review by matthewhelmke — COMMENTED** [review]
Looks good to me. I've noted a few small edits.

**Comment by pedroigor** [comment]
@matthewhelmke Sorry for the late reply, I have pushed another commit with your changes.

**Comment by matthewhelmke** [comment]
Thank you, @pedroigor. Looks good to me.
@stianst, this is ready for final review by someone on your team.

**Pull request merged**